### PR TITLE
feat: Timezone support for charts

### PR DESCRIPTION
**Pull request by timgl**
## Problem

I want filtering and day/week/month intervals to be based on my project timezone, not UTC

## Changes

- For every insight, convert all timestamps to whatever timezone is set
- Add a backend feature flag so we can roll this out safely

For us to roll this out properly we'll need to clear the cache of teams that have a different timezone set

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Unit tests
Manual QA. Set timezone in project and go to chart

### CODE REVIEW

File: posthog/tasks/update_cache.py [hunk 1]
```diff
@@ -149,19 +150,17 @@ def dashboard_item_update_task_params(
     return cache_key, cache_type, payload
 
 
-def _calculate_by_filter(filter: FilterType, key: str, team_id: int, cache_type: CacheType) -> List[Dict[str, Any]]:
+def _calculate_by_filter(filter: FilterType, key: str, team: Team, cache_type: CacheType) -> List[Dict[str, Any]]:
     insight_class = CACHE_TYPE_TO_INSIGHT_CLASS[cache_type]
 
     if cache_type == CacheType.PATHS:
-        result = insight_class(filter, Team(pk=team_id)).run(filter, Team(pk=team_id))
```
Comment (timgl) [inline posthog/tasks/update_cache.py:156]: @EDsCODE I think you added this `Team(pk=team_id)` workaround, which breaks actually looking up the right timezone. Any reason you did this?

Comment (EDsCODE) [reply posthog/tasks/update_cache.py:156]: I believe this was originally added to just avoid doing another lookup of team

File: posthog/models/team.py [hunk 2]
```diff
@@ -181,6 +182,17 @@ def get_effective_membership_level(self, user_id: int) -> Optional["Organization
...
 
+    def timezone_for_charts(self) -> str:
+        """
+        Stopgap function for rolling this feature out
+        """
+        distinct_id = self.organization.members.filter(is_active=True).first().distinct_id
+        if posthoganalytics.feature_enabled(
+            "timezone-for-charts", distinct_id, groups={"organization": self.organization_id}
+        ):
+            return self.timezone
```
Comment (EDsCODE) [inline posthog/models/team.py:193]: Where is this set?

File: ee/clickhouse/sql/events.py [hunk 3]
```diff
@@ -256,7 +256,7 @@
 
 NULL_SQL = """
 -- Creates zero values for all date axis ticks for the given date_from, date_to range
-SELECT toUInt16(0) AS total, {trunc_func}(toDateTime(%(date_to)s) - {interval_func}(number)) AS day_start
+SELECT toUInt16(0) AS total, {trunc_func}(toDateTime(%(date_to)s, %(timezone)s) - {interval_func}(number)) AS day_start
```
Comment (EDsCODE) [inline ee/clickhouse/sql/events.py:259]: We need to update the UI to communicate that the timezones are aligned with their local timezones now and not UTC

Comment (timgl) [reply ee/clickhouse/sql/events.py:259]: I wonder if we do or whether it'll just add noise. This functionality is behind a feature flag anyway so we can do this when we roll it out.

File: posthog/models/team.py (hunk 2 above)
Comment (timgl) [reply posthog/models/team.py:193]: It's the timezone field you can set on the team settings page.
image

**Comment by timgl** [comment]
Gonna merge this as no major blockers, will slowly roll this out to ourselves first.

**Pull request merged**